haystack/backends/elasticsearch_backend.py

```diff
@@ -731,6 +757,8 @@ def matching_all_fragment(self):
 
     def add_spatial(self, lat, lon, sfield, distance, filter='bbox'):
         """Adds spatial query parameters to search query"""
+        if six.PY3:
+            long = int
         kwargs = {
             'lat': lat,
             'long': long,
```

Comment:
I'm working on fixing the backend code now and updating the tests since we obviously didn't even enough coverage – ElasticSearch uses `lon` consistently and Solr appears to only use unnamed point pairs.